plone/app/contentlisting/catalog.py

```diff
@@ -39,12 +39,13 @@ def __getattr__(self, name):
 
         if name.startswith('_'):
             raise AttributeError(name)
-        if hasattr(aq_base(self._brain), name):
-            return getattr(self._brain, name)
-        elif hasattr(aq_base(self.getObject()), name):
-            return getattr(aq_base(self.getObject()), name)
-        else:
-            raise AttributeError(name)
+        brain_name = getattr(aq_base(self._brain), name, None)
+        if brain_name is not None:
+            return brain_name
+        object_name = getattr(aq_base(self.getObject()), name, None)
```

Comment:
It might be that an object _has_ a particular attribute but that the value is `None`.

That's why we typically write code like this:
```python
missing = object()
value = getattr(some_obj, name, missing)
if value is not missing:
    return value
```
You can use any value for `missing` – as long as you know that `some_obj.name` will never return that particular value. It's a "marker" for the case where the value is missing.

For performance reasons, it's best to put the `missing = object()` assignment out to module level.